docs/api-reference/data-fetching/getStaticProps.md

```diff
@@ -0,0 +1,210 @@
+---
+description: API reference for `getStaticProps`. Learn how to use `getStaticProps` to generate static pages with Next.js.
+---
+
+# `getStaticProps`
+
+<details>
+  <summary><b>Version History</b></summary>
+
+| Version   | Changes                                                                                                           |
+| --------- | ----------------------------------------------------------------------------------------------------------------- |
+| `v10.0.0` | `locale`, `locales`, `defaultLocale`, and `notFound` options added.                                               |
+| `v9.5.0`  | Stable [Incremental Static Regeneration](https://nextjs.org/blog/next-9-5#stable-incremental-static-regeneration) |
+| `v9.3.0`  | `getStaticProps` introduced.                                                                                      |
+
+</details>
+
+Exporting an `async` function called `getStaticProps` will pre-render a page at build time using the props returned from the function:
+
+```jsx
+export async function getStaticProps(context) {
+  return {
+    props: {}, // will be passed to the page component as props
+  }
+}
+```
+
+You can import modules in top-level scope for use in `getStaticProps`.
+Imports used will **not be bundled for the client-side**. This means you can write **server-side code directly in `getStaticProps`**, including fetching data from your database.
+
+If you are calling an external `API`, or fetching data from a database or Content Management System (CMS) in an `API` route, you can move that logic to `getStaticProps`, as both `API` routes and `getStaticProps` run on the server.
+
+Calling an `API` route from within `getStaticProps` can result in an additional call, reducing performance.
+
+Alternatively, if you are **not** using `API` routes to fetch data, then the [`fetch()`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) `API` _can_ be used directly in `getStaticProps` to fetch data.
```

Comment:
I see this changed since https://github.com/vercel/next.js/pull/30615/files#r764824267

Could this have a heading for easier linking? Seems to be a popular mistake.

Reply:
I had duplicate content around this subject, so have consolidated it in `basic-features/data-fetching/getStaticProps#write-server-side-code-directly`